core/pkg/ingress/annotations/authtls/main.go

```diff
@@ -30,6 +29,13 @@ const (
 	authTLSSecret = "ingress.kubernetes.io/auth-tls-secret"
 )
 
+// AuthCertificate has a method that searchs for a secret
+// that contains a SSL certificate.
+// The secret must contain 3 keys named:
+type AuthCertificate interface {
```

Comment:
Should this also be in `resolver` for consistency (`resolver.AuthCertificate` to match `resolver.Secret`)?